### PR TITLE
SPCTR-2508 Dynamic Content Optimization and Data Fetch Approach Changes

### PR DESCRIPTION
**Pull request by abhishekrijal**
### Description
Changed dynamic content data attribute storage format and tag names. Removed unnecessary slugs and used the getEntityRecord function to fetch the following data of current-post.

- Post ID
- Post Excerpt
- Post Title

The changes are a part of backend optimizations and will break in the front end. Also, added compatibility with the new loop builder.

### Types of changes
Breaking change

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/advanced-heading/attributes.js [hunk 1]
```diff
@@ -39,8 +39,7 @@ const attributes = {
 		default: true,
 	},
 	headingTitle: {
-		source: 'html',
-		selector: '.uagb-heading-text',
+		type: 'string',
```
Comment (RajkiranBagal) [inline src/blocks/advanced-heading/attributes.js:42]: This may cause backward compatibility issues?

File: src/blocks/advanced-heading/render.js [hunk 2]
```diff
@@ -1,27 +1,26 @@
-import classnames from 'classnames';
-import { RichText } from '@wordpress/block-editor';
-import { __ } from '@wordpress/i18n';
 import { memo } from '@wordpress/element';
+import classnames from 'classnames';
```
Comment (RajkiranBagal) [inline src/blocks/advanced-heading/render.js:2]: We can use `uagbClassNames`

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: src/blocks/advanced-heading/render.js (hunk 2 above)
Comment (abhishekrijal) [reply src/blocks/advanced-heading/render.js:2]: updated

File: src/blocks/advanced-heading/attributes.js (hunk 1 above)
Comment (abhishekrijal) [reply src/blocks/advanced-heading/attributes.js:42]: @rahulvarma722 Check this, did we make this change for the filter? verify if it is required.

**Review by rahulvarma722 — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**